File: Desafios_Javascript/Desenvolvimento_de_problemas_avançados_em_Javascript/desafio_01.js
/* Desafio 1 - Ordenando Números Pares e Ímpares */

let nEntries = parseInt(gets());
let par = [];
let imp = [];

for(let counter = 0; counter < nEntries; counter++){
  let input = parseInt(gets());
  
  if(input%2 == 0){
    par.push(input);
    
    par.sort((a, b) => {
      if(a < b) return -1;
      if(a > b) return 1;
      if(a == b) return 0;
    });

  }else{
    imp.push(input);
    
    imp.sort((a, b) => {
      if(a < b) return 1;
      if(a > b) return -1;
      if(a == b) return 0;
    });
  }
}

par.forEach(num => console.log(num));
imp.forEach(num => console.log(num));

File: Desafios_Java/Ordenação_e_Filtros_em_Java/desafio_01.java
/* Desafio 1 - Ordenando Números Pares e Ímpares*/

package Desafios_Java.Ordenação_e_Filtros_em_Java;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;

public class desafio_01 {
        public static void main(String[] args) throws IOException{

        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        int nEntries = Integer.parseInt(reader.readLine());

        int[] par = {};
        int[] imp = {};

        for(int counter = 0; counter < nEntries; counter++){
            
            int input = Integer.parseInt(reader.readLine());

            if(input%2 == 0){
                par = Arrays.copyOf(par, par.length + 1);
                par[par.length-1] = input;
                Arrays.sort(par);
            }
            else{
                imp = Arrays.copyOf(imp, imp.length + 1);
                imp[imp.length-1] = input;
                Arrays.sort(imp);
            }
        }

        for(int pos = 0; pos < par.length; pos++){
            System.out.println(par[pos]);
        }
        for(int pos = imp.length-1; pos >= 0; pos--){
            System.out.println(imp[pos]);
        }
    }
}